src/main/java/uk/ac/ebi/eva/utils/VcfNumberOfLinesEstimator.java

```diff
@@ -0,0 +1,148 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.utils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import uk.ac.ebi.eva.pipeline.listeners.StepProgressListener;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Scanner;
+import java.util.zip.GZIPInputStream;
+
+/**
+ * Estimate the number of lines in a VCF file.
+ * <p>
+ * Given that the VCF file could be VERY big then we estimate the number of lines using the following steps:
+ * 1) retrieve the size in bytes of the whole zipped VCF (vcfFileSize)
+ * 2) retrieve the size in bytes of the head of the VCF in a zipped file (vcfHeadFileSize)
+ * 3) retrieve the size in bytes of the first 100 lines of the VCF in a zipped file and divide the value by 100 to
+ * obtain the size of a single line (singleVcfLineSize)
+ * 4) calculate the (vcfFileSize - vcfHeadFileSize) / singleVcfLineSize to estimate the total number of line in the VCF.
+ * <p>
+ * Why 100 in point 3?
+ * Tested on a VCF with 157049 lines, 100 is the best and minimum number of lines to compress. This should generate an
+ * estimated total number of lines similar to the real one.
+ * <p>
+ * In case of small VCF the NUMBER_OF_LINES will be 0 and no % will be printed in {@link StepProgressListener#afterChunk}
+ */
+public class VcfNumberOfLinesEstimator {
+    private static final Logger logger = LoggerFactory.getLogger(VcfNumberOfLinesEstimator.class);
+
+    private static final int NUMBER_OF_LINES = 100;
+
+    /**
+     * Given that the VCF file could be VERY big then we estimate the number of lines using the following steps:
+     * 1) retrieve the size in bytes of the whole zipped VCF (vcfFileSize)
+     * 2) retrieve the size in bytes of the head of the VCF in a zipped file (vcfHeadFileSize)
+     * 3) retrieve the size in bytes of the first 100 lines in the body of the VCF in a zipped file and divide the
+     * value by 100 to obtain the size of a single line (singleVcfLineSize)
+     * 4) calculate the (vcfFileSize - vcfHeadFileSize) / singleVcfLineSize to estimate the total number of line in the VCF.
+     *
+     * Why 100 in point 3?
+     * Tested on a VCF with 157049 lines, 100 is the best and minimum number of lines to compress. This should generate an
+     * estimated total number of lines similar to the real one.
+     * <p>
+     * In case of small VCF the NUMBER_OF_LINES will be 0 and no % will be printed in {@link StepProgressListener#afterChunk}
+     */
```

Comment:
If this is documenting a method it shouldn't be duplicated in the class documentation. Also please document the input and the output using the appropriate Javadoc tags.